packages/peregrine/lib/talons/ProductFullDetail/useProductFullDetail.js

```diff
@@ -163,10 +166,35 @@ export const useProductFullDetail = props => {
         productType
     );
 
-    const [{ isAddingItem }, { addItemToCart }] = useCartContext();
+    const [{ cartId, isAddingItem }, { addItemToCart }] = useCartContext();
+    // This value should always be a valid cart id.
+    console.log(`pfd cartId`, cartId);
 
+    const apolloClient = useApolloClient();
     const [addConfigurableProductToCart] = useMutation(
-        addConfigurableProductToCartMutation
+        addConfigurableProductToCartMutation,
+        {
+            onError: async err => {
+                console.log('got error', err);
+                // Refetch cart id when the mutation fails. I was hoping this
+                // could be done in CartContext but the lazy query is not being
+                // triggered. An alternative may be to do this in a link?
+                try {
+                    // We could specify to only re-create cart id for specific
+                    // errors.
+                    const { data } = await apolloClient.mutate({
+                        mutation: createCartMutation
+                    });
+                    console.log('Replacing local cart id!');
+                    apolloClient.writeQuery({
+                        query: GET_LOCAL_CART_ID,
+                        data
+                    });
```

Comment:
Does it make sense to recreate the cart id like this in every mutation/query where the cart id may be invalidated? Feels as if it could get unwieldy. Would prefer to do it somewhere more global like the cart context resolver.